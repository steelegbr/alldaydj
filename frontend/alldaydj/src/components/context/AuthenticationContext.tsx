import React, { useEffect } from 'react'
import {
  getAuthenticationStatusFromLocalStorage,
  refreshAccessToken
} from '../../services/AuthenticationService'
import { getLogger } from '../../services/LoggingService'

export type AuthenticationStage =
  | 'Unauthenticated'
  | 'AccessTokenRefreshNeeded'
  | 'RefreshingAccessToken'
  | 'Authenticated';

export interface AuthenticationStatus {
  stage: AuthenticationStage;
  accessToken?: string;
  refreshToken?: string;
  accessTokenExpiry?: Date;
  refreshTokenExpiry?: Date;
  tenant?: string;
}

export interface AuthenticationStatusProps {
  authenticationStatus: AuthenticationStatus;
  setAuthenticationStatus: React.Dispatch<React.SetStateAction<AuthenticationStatus>>;
}

export const AuthenticationContext = React.createContext<undefined | AuthenticationStatusProps>(
  undefined
)

interface AuthenticationProviderProps {
  children: React.ReactElement;
}

export const AuthenticationProvider = ({ children }: AuthenticationProviderProps): React.ReactElement => {
  const [authenticationStatus, setAuthenticationStatus] = React.useState<AuthenticationStatus>(
    getAuthenticationStatusFromLocalStorage()
  )

  useEffect(() => {
    const interval = setInterval(() => {
      const log = getLogger()
      log.debug('Checking current authentication status.')

      const newAuthStatus = getAuthenticationStatusFromLocalStorage()
      if (newAuthStatus.stage !== authenticationStatus.stage) {
        log.info(`Change of authentication stage to ${newAuthStatus.stage}.`)
        setAuthenticationStatus(newAuthStatus)
      }
    }, Number(process.env.REACT_APP_AUTH_INTERVAL))

    return () => clearInterval(interval)
  }, [])

  useEffect(() => {
    const log = getLogger()
    if (authenticationStatus.stage === 'AccessTokenRefreshNeeded') {
      refreshAccessToken(authenticationStatus.refreshToken || '', log, setAuthenticationStatus)
    }
  }, [authenticationStatus])

  return (
    <AuthenticationContext.Provider value={{ authenticationStatus, setAuthenticationStatus }}>
      {children}
    </AuthenticationContext.Provider>
  )
}
