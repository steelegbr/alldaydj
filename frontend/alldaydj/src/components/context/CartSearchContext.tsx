import React from 'react';
import { useLocation } from 'react-router-dom';
import { CartSearchConditions } from '../../api/models/Search';
import { cartSearchContextFromQueryString } from '../../services/SearchService';

export type CartSearchStatus = 'NotStarted' | 'ReadyToSearch' | 'Searching' | 'ResultsReturned' | 'Error';

export interface CartSearch {
    conditions: CartSearchConditions,
    status: CartSearchStatus
}

interface CartSearchContextType {
    search: CartSearch,
    setSearch: React.Dispatch<React.SetStateAction<CartSearch>>
}

const CartSearchContext = React.createContext<CartSearchContextType>({
  search: {
    conditions: {
      advanced: 'false',
      artist: '',
      page: '1',
      resultsPerPage: '10',
      search: '',
      title: '',
    },
    status: 'NotStarted',
  },
  setSearch: () => {},
});

export interface CartSearchProviderProps {
    children: React.ReactElement;
  }

export const CartSearchProvider = ({ children }: CartSearchProviderProps) => {
  const query = new URLSearchParams(useLocation().search);
  const [search, setSearch] = React.useState<CartSearch>(
    cartSearchContextFromQueryString(query),
  );

  return (
    <CartSearchContext.Provider value={{ search, setSearch }}>
      {children}
    </CartSearchContext.Provider>
  );
};
export default CartSearchContext;
