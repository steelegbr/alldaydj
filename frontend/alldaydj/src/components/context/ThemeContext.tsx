import { createMuiTheme, MuiThemeProvider } from '@material-ui/core/styles'
import React from 'react'
import { getLogger } from '../../services/LoggingService'

export interface ThemeSettings {
  darkMode: boolean;
}

export interface ThemeContextProps {
  themeSettings: ThemeSettings;
  setThemeSettings: React.Dispatch<React.SetStateAction<ThemeSettings>>;
}

export const ThemeContext = React.createContext<undefined | ThemeContextProps>(undefined)

interface ThemeProviderProps {
  children: React.ReactElement;
}

export const ThemeProvider = ({ children }: ThemeProviderProps): React.ReactElement => {
  const logger = getLogger()
  const darkMode = localStorage.getItem('darkMode') === 'true'
  logger.info(`Dark mode setting from local storage: ${darkMode}.`)

  const [themeSettings, setThemeSettings] = React.useState<ThemeSettings>({
    darkMode: darkMode
  })

  const theme = createMuiTheme({
    palette: {
      type: themeSettings.darkMode ? 'dark' : 'light'
    }
  })

  localStorage.setItem('darkMode', themeSettings.darkMode ? 'true' : 'false')
  logger.info(`Set dark mode to ${themeSettings.darkMode} in local storage.`)

  return (
    <ThemeContext.Provider value={{ themeSettings, setThemeSettings }}>
      <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>
    </ThemeContext.Provider>
  )
}
