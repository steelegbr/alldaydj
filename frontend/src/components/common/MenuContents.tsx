/**
    AllDay DJ - Radio Automation
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import {
  Divider, List, ListItem, ListItemIcon, ListItemText,
} from '@mui/material';
import makeStyles from '@mui/styles/makeStyles';
import createStyles from '@mui/styles/createStyles';
import { Theme } from '@mui/material/styles';
import { ExitToApp, LibraryMusic } from '@mui/icons-material';
import React from 'react';
import { useHistory } from 'react-router-dom';
import Paths from 'routing/Paths';

const useStyles = makeStyles((theme: Theme) => createStyles({
  toolbar: theme.mixins.toolbar,
}));

const MenuContents = (): React.ReactElement => {
  const classes = useStyles();
  const history = useHistory();

  const menuItemLibrary = () => (
    <ListItem
      button
      data-test="button-library"
      key="Music Library"
      onClick={(event) => {
        event.preventDefault();
        history.push(Paths.library.search);
      }}
    >
      <ListItemIcon>
        <LibraryMusic />
      </ListItemIcon>
      <ListItemText primary="Music Library" />
    </ListItem>
  );

  const menuItemLogout = () => (
    <ListItem
      button
      data-test="button-logout"
      key="Log Out"
      onClick={(event) => {
        event.preventDefault();
        history.push(Paths.auth.logout);
      }}
    >
      <ListItemIcon>
        <ExitToApp />
      </ListItemIcon>
      <ListItemText primary="Log Out" />
    </ListItem>
  );

  return (
    <div>
      <div className={classes.toolbar} />
      <Divider />
      <List>
        {menuItemLibrary()}
      </List>
      <Divider />
      <List>
        {menuItemLogout()}
      </List>
    </div>
  );
};

export default MenuContents;
