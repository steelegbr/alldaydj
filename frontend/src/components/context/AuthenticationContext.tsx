/**
    AllDay DJ - Radio Automation
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import React, { useEffect } from 'react';
import {
  getAuthenticationStatusFromLocalStorage,
  refreshAccessToken,
} from 'services/AuthenticationService';
import { getLogger } from 'services/LoggingService';

export type AuthenticationStage =
  | 'Unauthenticated'
  | 'AccessTokenRefreshNeeded'
  | 'RefreshingAccessToken'
  | 'Authenticated';

export interface AuthenticationStatus {
  stage: AuthenticationStage;
  accessToken?: string;
  refreshToken?: string;
  accessTokenExpiry?: Date;
  refreshTokenExpiry?: Date;
}

export interface AuthenticationStatusProps {
  authenticationStatus: AuthenticationStatus;
  setAuthenticationStatus: React.Dispatch<React.SetStateAction<AuthenticationStatus>>;
}

export const AuthenticationContext = React.createContext<undefined | AuthenticationStatusProps>(
  undefined,
);

export interface AuthenticationProviderProps {
  children: React.ReactElement;
}

const AuthenticationProvider = ({ children }: AuthenticationProviderProps): React.ReactElement => {
  const [authenticationStatus, setAuthenticationStatus] = React.useState<AuthenticationStatus>(
    getAuthenticationStatusFromLocalStorage(),
  );

  useEffect(() => {
    const interval = setInterval(() => {
      const log = getLogger();
      log.debug('Checking current authentication status.');

      const newAuthStatus = getAuthenticationStatusFromLocalStorage();
      if (newAuthStatus.stage !== authenticationStatus.stage) {
        log.info(`Change of authentication stage to ${newAuthStatus.stage}.`);
        setAuthenticationStatus(newAuthStatus);
      }
    }, Number(process.env.REACT_APP_AUTH_INTERVAL || '5000'));

    return () => clearInterval(interval);
  }, [authenticationStatus.stage]);

  useEffect(() => {
    if (authenticationStatus.stage === 'AccessTokenRefreshNeeded') {
      refreshAccessToken(authenticationStatus.refreshToken || '', setAuthenticationStatus);
    }
  }, [authenticationStatus]);

  return (
    <AuthenticationContext.Provider value={{ authenticationStatus, setAuthenticationStatus }}>
      {children}
    </AuthenticationContext.Provider>
  );
};

export default AuthenticationProvider;
