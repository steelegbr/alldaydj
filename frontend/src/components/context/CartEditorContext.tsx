/**
    AllDay DJ - Radio Automation
    Copyright (C) 2020-2021 Marc Steele

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <https://www.gnu.org/licenses/>.
*/

import { Cart } from 'api/models/Cart';
import React from 'react';

export interface CartEditorContextType {
    cart?: Cart,
    file?: File,
    setCart: React.Dispatch<React.SetStateAction<Cart | undefined>>;
    setFile: React.Dispatch<React.SetStateAction<File | undefined>>;
    clearCart: () => void;
    clearFile: () => void;
}

const noOp = () => {
  // Do nothing
};

export const CartEditorContext = React.createContext<CartEditorContextType>(
  {
    setCart: noOp,
    setFile: noOp,
    clearCart: noOp,
    clearFile: noOp,
  },
);

export interface CartEditorProviderProps {
    children: React.ReactElement;
}

export const CartEditorProvider = ({ children }: CartEditorProviderProps): React.ReactElement => {
  const [cart, setCart] = React.useState<Cart>();
  const [file, setFile] = React.useState<File>();

  const clearCart = () => {
    setCart(undefined);
  };

  const clearFile = () => {
    setFile(undefined);
  };

  return (
    <CartEditorContext.Provider value={{
      cart,
      setCart,
      clearCart,
      file,
      setFile,
      clearFile,
    }}
    >
      {children}
    </CartEditorContext.Provider>
  );
};
