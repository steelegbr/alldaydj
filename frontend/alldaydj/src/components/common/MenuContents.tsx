import {
  createStyles, Divider, List, ListItem, ListItemIcon, ListItemText, makeStyles, Theme,
} from '@material-ui/core';
import { Domain, ExitToApp, LibraryMusic } from '@material-ui/icons';
import React from 'react';
import { useHistory } from 'react-router-dom';
import { Paths } from '../../routing/Paths';
import { logOut } from '../../services/AuthenticationService';
import { AuthenticationContext } from '../context/AuthenticationContext';

const useStyles = makeStyles((theme: Theme) => createStyles({
  toolbar: theme.mixins.toolbar,
}));

const MenuContents = (): React.ReactElement => {
  const history = useHistory();
  const classes = useStyles();
  const authenticationContext = React.useContext(AuthenticationContext);
  const currentTenant = authenticationContext?.authenticationStatus.tenant;

  const doLogOut = () => {
    authenticationContext?.setAuthenticationStatus(logOut());
  };

  const changeTenant = (event: React.SyntheticEvent) => {
    event.preventDefault();
    history.push(Paths.auth.tenancy);
  };

  const menuItemLibrary = () => (
    <ListItem button key="Music Library">
      <ListItemIcon>
        <LibraryMusic />
      </ListItemIcon>
      <ListItemText primary="Music Library" />
    </ListItem>
  );

  const menuItemTenantChanger = () => (
    <ListItem button data-test="button-change-tenant" key={Paths.auth.tenancy} onClick={changeTenant}>
      <ListItemIcon>
        <Domain />
      </ListItemIcon>
      <ListItemText primary="Change Tenant" secondary={`Current: ${currentTenant}`} />
    </ListItem>
  );

  const menuItemLogout = () => (
    <ListItem button data-test="button-logout" key="Log Out" onClick={doLogOut}>
      <ListItemIcon>
        <ExitToApp />
      </ListItemIcon>
      <ListItemText primary="Log Out" />
    </ListItem>
  );

  return (
    <div>
      <div className={classes.toolbar} />
      <Divider />
      <List>
        {menuItemLibrary()}
      </List>
      <Divider />
      <List>
        {menuItemTenantChanger()}
        {menuItemLogout()}
      </List>
    </div>
  );
};

export default MenuContents;
