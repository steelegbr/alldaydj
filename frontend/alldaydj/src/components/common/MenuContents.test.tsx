import React from 'react';
import { mount } from 'enzyme';
import MenuContents from './MenuContents';
import { AuthenticationContext, AuthenticationStatusProps } from '../context/AuthenticationContext';

const mockPush = jest.fn();
const mockSetAuthStatus = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockPush,
  }),
}));

describe('menu contents', () => {
  const mountComponent = () => {
    const contextValue : AuthenticationStatusProps = {
      authenticationStatus: {
        stage: 'Authenticated',
      },
      setAuthenticationStatus: mockSetAuthStatus,
    };
    return mount(
      <AuthenticationContext.Provider value={contextValue}>
        <MenuContents />
      </AuthenticationContext.Provider>,
    );
  };

  it('snapshot', () => {
    const component = mountComponent();
    expect(component).toMatchSnapshot();
  });

  it('log out', () => {
    const component = mountComponent();
    const logoutButton = component.find("[data-test='button-logout']").first();
    logoutButton.simulate('click');
    expect(mockSetAuthStatus).toBeCalledWith({ stage: 'Unauthenticated' });
  });
});
