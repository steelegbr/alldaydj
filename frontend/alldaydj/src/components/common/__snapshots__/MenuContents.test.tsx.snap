// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`menu contents snapshot 1`] = `
<MenuContents>
  <div>
    <div
      className="makeStyles-toolbar-1"
    />
    <WithStyles(ForwardRef(Divider))>
      <ForwardRef(Divider)
        classes={
          Object {
            "absolute": "MuiDivider-absolute",
            "flexItem": "MuiDivider-flexItem",
            "inset": "MuiDivider-inset",
            "light": "MuiDivider-light",
            "middle": "MuiDivider-middle",
            "root": "MuiDivider-root",
            "vertical": "MuiDivider-vertical",
          }
        }
      >
        <hr
          className="MuiDivider-root"
        />
      </ForwardRef(Divider)>
    </WithStyles(ForwardRef(Divider))>
    <WithStyles(ForwardRef(List))>
      <ForwardRef(List)
        classes={
          Object {
            "dense": "MuiList-dense",
            "padding": "MuiList-padding",
            "root": "MuiList-root",
            "subheader": "MuiList-subheader",
          }
        }
      >
        <ul
          className="MuiList-root MuiList-padding"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            data-test="button-library"
            key="Music Library"
            onClick={[Function]}
          >
            <ForwardRef(ListItem)
              button={true}
              classes={
                Object {
                  "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                  "button": "MuiListItem-button",
                  "container": "MuiListItem-container",
                  "dense": "MuiListItem-dense",
                  "disabled": "Mui-disabled",
                  "divider": "MuiListItem-divider",
                  "focusVisible": "Mui-focusVisible",
                  "gutters": "MuiListItem-gutters",
                  "root": "MuiListItem-root",
                  "secondaryAction": "MuiListItem-secondaryAction",
                  "selected": "Mui-selected",
                }
              }
              data-test="button-library"
              onClick={[Function]}
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiListItem-root MuiListItem-gutters MuiListItem-button"
                component="div"
                data-test="button-library"
                disabled={false}
                focusVisibleClassName="Mui-focusVisible"
                onClick={[Function]}
              >
                <ForwardRef(ButtonBase)
                  className="MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="div"
                  data-test="button-library"
                  disabled={false}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                >
                  <div
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    data-test="button-library"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <WithStyles(ForwardRef(ListItemIcon))
                      key=".0"
                    >
                      <ForwardRef(ListItemIcon)
                        classes={
                          Object {
                            "alignItemsFlexStart": "MuiListItemIcon-alignItemsFlexStart",
                            "root": "MuiListItemIcon-root",
                          }
                        }
                      >
                        <div
                          className="MuiListItemIcon-root"
                        >
                          <ForwardRef(LibraryMusicIcon)>
                            <WithStyles(ForwardRef(SvgIcon))>
                              <ForwardRef(SvgIcon)
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 5h-3v5.5c0 1.38-1.12 2.5-2.5 2.5S10 13.88 10 12.5s1.12-2.5 2.5-2.5c.57 0 1.08.19 1.5.51V5h4v2zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef(LibraryMusicIcon)>
                        </div>
                      </ForwardRef(ListItemIcon)>
                    </WithStyles(ForwardRef(ListItemIcon))>
                    <WithStyles(ForwardRef(ListItemText))
                      key=".1"
                      primary="Music Library"
                    >
                      <ForwardRef(ListItemText)
                        classes={
                          Object {
                            "dense": "MuiListItemText-dense",
                            "inset": "MuiListItemText-inset",
                            "multiline": "MuiListItemText-multiline",
                            "primary": "MuiListItemText-primary",
                            "root": "MuiListItemText-root",
                            "secondary": "MuiListItemText-secondary",
                          }
                        }
                        primary="Music Library"
                      >
                        <div
                          className="MuiListItemText-root"
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiListItemText-primary"
                            component="span"
                            display="block"
                            variant="body1"
                          >
                            <ForwardRef(Typography)
                              className="MuiListItemText-primary"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              component="span"
                              display="block"
                              variant="body1"
                            >
                              <span
                                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                              >
                                Music Library
                              </span>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </div>
                      </ForwardRef(ListItemText)>
                    </WithStyles(ForwardRef(ListItemText))>
                    <WithStyles(memo)
                      center={false}
                    >
                      <ForwardRef(TouchRipple)
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(memo)>
                  </div>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(ListItem)>
          </WithStyles(ForwardRef(ListItem))>
        </ul>
      </ForwardRef(List)>
    </WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(Divider))>
      <ForwardRef(Divider)
        classes={
          Object {
            "absolute": "MuiDivider-absolute",
            "flexItem": "MuiDivider-flexItem",
            "inset": "MuiDivider-inset",
            "light": "MuiDivider-light",
            "middle": "MuiDivider-middle",
            "root": "MuiDivider-root",
            "vertical": "MuiDivider-vertical",
          }
        }
      >
        <hr
          className="MuiDivider-root"
        />
      </ForwardRef(Divider)>
    </WithStyles(ForwardRef(Divider))>
    <WithStyles(ForwardRef(List))>
      <ForwardRef(List)
        classes={
          Object {
            "dense": "MuiList-dense",
            "padding": "MuiList-padding",
            "root": "MuiList-root",
            "subheader": "MuiList-subheader",
          }
        }
      >
        <ul
          className="MuiList-root MuiList-padding"
        >
          <WithStyles(ForwardRef(ListItem))
            button={true}
            data-test="button-logout"
            key="Log Out"
            onClick={[Function]}
          >
            <ForwardRef(ListItem)
              button={true}
              classes={
                Object {
                  "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                  "button": "MuiListItem-button",
                  "container": "MuiListItem-container",
                  "dense": "MuiListItem-dense",
                  "disabled": "Mui-disabled",
                  "divider": "MuiListItem-divider",
                  "focusVisible": "Mui-focusVisible",
                  "gutters": "MuiListItem-gutters",
                  "root": "MuiListItem-root",
                  "secondaryAction": "MuiListItem-secondaryAction",
                  "selected": "Mui-selected",
                }
              }
              data-test="button-logout"
              onClick={[Function]}
            >
              <WithStyles(ForwardRef(ButtonBase))
                className="MuiListItem-root MuiListItem-gutters MuiListItem-button"
                component="div"
                data-test="button-logout"
                disabled={false}
                focusVisibleClassName="Mui-focusVisible"
                onClick={[Function]}
              >
                <ForwardRef(ButtonBase)
                  className="MuiListItem-root MuiListItem-gutters MuiListItem-button"
                  classes={
                    Object {
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "root": "MuiButtonBase-root",
                    }
                  }
                  component="div"
                  data-test="button-logout"
                  disabled={false}
                  focusVisibleClassName="Mui-focusVisible"
                  onClick={[Function]}
                >
                  <div
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                    data-test="button-logout"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <WithStyles(ForwardRef(ListItemIcon))
                      key=".0"
                    >
                      <ForwardRef(ListItemIcon)
                        classes={
                          Object {
                            "alignItemsFlexStart": "MuiListItemIcon-alignItemsFlexStart",
                            "root": "MuiListItemIcon-root",
                          }
                        }
                      >
                        <div
                          className="MuiListItemIcon-root"
                        >
                          <ForwardRef(ExitToAppIcon)>
                            <WithStyles(ForwardRef(SvgIcon))>
                              <ForwardRef(SvgIcon)
                                classes={
                                  Object {
                                    "colorAction": "MuiSvgIcon-colorAction",
                                    "colorDisabled": "MuiSvgIcon-colorDisabled",
                                    "colorError": "MuiSvgIcon-colorError",
                                    "colorPrimary": "MuiSvgIcon-colorPrimary",
                                    "colorSecondary": "MuiSvgIcon-colorSecondary",
                                    "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                    "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                    "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                    "root": "MuiSvgIcon-root",
                                  }
                                }
                              >
                                <svg
                                  aria-hidden={true}
                                  className="MuiSvgIcon-root"
                                  focusable="false"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                                  />
                                </svg>
                              </ForwardRef(SvgIcon)>
                            </WithStyles(ForwardRef(SvgIcon))>
                          </ForwardRef(ExitToAppIcon)>
                        </div>
                      </ForwardRef(ListItemIcon)>
                    </WithStyles(ForwardRef(ListItemIcon))>
                    <WithStyles(ForwardRef(ListItemText))
                      key=".1"
                      primary="Log Out"
                    >
                      <ForwardRef(ListItemText)
                        classes={
                          Object {
                            "dense": "MuiListItemText-dense",
                            "inset": "MuiListItemText-inset",
                            "multiline": "MuiListItemText-multiline",
                            "primary": "MuiListItemText-primary",
                            "root": "MuiListItemText-root",
                            "secondary": "MuiListItemText-secondary",
                          }
                        }
                        primary="Log Out"
                      >
                        <div
                          className="MuiListItemText-root"
                        >
                          <WithStyles(ForwardRef(Typography))
                            className="MuiListItemText-primary"
                            component="span"
                            display="block"
                            variant="body1"
                          >
                            <ForwardRef(Typography)
                              className="MuiListItemText-primary"
                              classes={
                                Object {
                                  "alignCenter": "MuiTypography-alignCenter",
                                  "alignJustify": "MuiTypography-alignJustify",
                                  "alignLeft": "MuiTypography-alignLeft",
                                  "alignRight": "MuiTypography-alignRight",
                                  "body1": "MuiTypography-body1",
                                  "body2": "MuiTypography-body2",
                                  "button": "MuiTypography-button",
                                  "caption": "MuiTypography-caption",
                                  "colorError": "MuiTypography-colorError",
                                  "colorInherit": "MuiTypography-colorInherit",
                                  "colorPrimary": "MuiTypography-colorPrimary",
                                  "colorSecondary": "MuiTypography-colorSecondary",
                                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                  "displayBlock": "MuiTypography-displayBlock",
                                  "displayInline": "MuiTypography-displayInline",
                                  "gutterBottom": "MuiTypography-gutterBottom",
                                  "h1": "MuiTypography-h1",
                                  "h2": "MuiTypography-h2",
                                  "h3": "MuiTypography-h3",
                                  "h4": "MuiTypography-h4",
                                  "h5": "MuiTypography-h5",
                                  "h6": "MuiTypography-h6",
                                  "noWrap": "MuiTypography-noWrap",
                                  "overline": "MuiTypography-overline",
                                  "paragraph": "MuiTypography-paragraph",
                                  "root": "MuiTypography-root",
                                  "srOnly": "MuiTypography-srOnly",
                                  "subtitle1": "MuiTypography-subtitle1",
                                  "subtitle2": "MuiTypography-subtitle2",
                                }
                              }
                              component="span"
                              display="block"
                              variant="body1"
                            >
                              <span
                                className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                              >
                                Log Out
                              </span>
                            </ForwardRef(Typography)>
                          </WithStyles(ForwardRef(Typography))>
                        </div>
                      </ForwardRef(ListItemText)>
                    </WithStyles(ForwardRef(ListItemText))>
                    <WithStyles(memo)
                      center={false}
                    >
                      <ForwardRef(TouchRipple)
                        center={false}
                        classes={
                          Object {
                            "child": "MuiTouchRipple-child",
                            "childLeaving": "MuiTouchRipple-childLeaving",
                            "childPulsate": "MuiTouchRipple-childPulsate",
                            "ripple": "MuiTouchRipple-ripple",
                            "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                            "rippleVisible": "MuiTouchRipple-rippleVisible",
                            "root": "MuiTouchRipple-root",
                          }
                        }
                      >
                        <span
                          className="MuiTouchRipple-root"
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            component={null}
                            exit={true}
                          />
                        </span>
                      </ForwardRef(TouchRipple)>
                    </WithStyles(memo)>
                  </div>
                </ForwardRef(ButtonBase)>
              </WithStyles(ForwardRef(ButtonBase))>
            </ForwardRef(ListItem)>
          </WithStyles(ForwardRef(ListItem))>
        </ul>
      </ForwardRef(List)>
    </WithStyles(ForwardRef(List))>
  </div>
</MenuContents>
`;
